pkg/webhook/v1beta1/pod/inject_webhook.go

```diff
@@ -140,15 +141,37 @@ func (s *SidecarInjector) Mutate(pod *v1.Pod, namespace string) (*v1.Pod, error)
 	// Add Katib Trial labels to the Pod metadata.
 	mutatePodMetadata(mutatedPod, trial)
 
+	// Pass env variable KATIB_TRIAL_NAME to the primary container using fieldPath.
+	index := getPrimaryContainerIndex(mutatedPod.Spec.Containers, trial.Spec.PrimaryContainerName)
+	if index >= 0 {
```

Comment:
I am also wondering, do we really need to set `primaryPodLabels` and `primaryContainerName` when metrics collector is Push ? @Electronic-Waste @johnugeorge @tenzen-y ?

Reply:
I think we don't need to set `primaryPodLabels` and `primaryContainerName` for push-based metrics collector. 

But I guess it's not worthwhile to add more if-else branches since it will sacrifice readability of code. If we prohibit specifying `primaryPodLabel` and `primaryContainerName` for the Trial, users will find that it's not similar to other collectors' specification and start figuring out why throughout docs, github and stackoverflow, which will definitely do harm to their experience of using Katib.

So I suggest using unified interface to maintain backward compatibility and users' experience.

Reply:
>users will find that it's not similar to other collectors' specifications when using kubectl to apply Experiment

Actually, I think I am wrong. We also use the `primaryContainerName` to mutate Suggestion's volume to the Trial: https://github.com/kubeflow/katib/blob/master/pkg/webhook/v1beta1/pod/inject_webhook.go#L334.
So we might need to keep it for `push` metrics collector as well.